### ISSUE TITLE
Eventlist: Option "shortened view" not activated – multi-day events show up with dates already past

### ISSUE THREAD

**Issue by arno1979** [contao/core#8782]
Steps to reproduce:
- An eventlist module configured to show only upcoming events.
- Option "shortened view" NOT activated.
- A multi-day event that begins in the past (e.g. 2017-09-01) and reaches into the future (e.g. 2017-09-15)

Result: Today (2017-09-11) the eventlist shows single events from 2017-09-01 to 2017-09-15. (see screenshot)

Expected result: Only events from 2017-09-11 to 2017-09-15 are shown.

This applies to both Contao 3.x.x and 4.x.x

<img width="185" alt="bildschirmfoto 2017-09-11 um 15 55 04" src="https://user-images.githubusercontent.com/1838113/30278147-fda6a82e-9709-11e7-82dc-c6794415f7f1.png">

**Comment by leofeyer** [comment]
Does the "hide running events" checkbox change this?

**Comment by arno1979** [comment]
No, it doesn't make a difference. (Tested on the demo just now.)

**Comment by leofeyer** [comment]
@fritzmg Can you confirm this?

**Comment by arno1979** [comment]
Working further on this and also #8786 I found out the following:
In `ModuleEventlist.php:
`Adding a value `$currTime = \Date::floorToMinute();` on line 160 (inspiration taken from [calendar_extended](https://github.com/kmielke/calendar_extended)) and adding two if statements comparing the current timestamp against the correct timestamp of each single event a little further below in the code helped to seemingly solve the problem for me.

My changes look like this:
On line 174 we find this if statement:
```php
// Skip occurrences in the past but show running events (see #8497)
if ($event['repeatEnd'] && $event['end'] < $intStart)
{
    continue;
}
```
I added the two following if statements:
```php
// Skip occurrences in the past but show running events, for long list view only
if (!$this->cal_noSpan && strpos($this->cal_format,'next') !== false && $day < $currTime-86400)
{
    continue;
}

// Skip occurrences in the future but show running events, for long list view only
if (!$this->cal_noSpan && strpos($this->cal_format,'past') !== false && $day > $currTime-86400)
{
    continue;
}
```

I have not tested this extensively and I admit that it does not look very elegant. But at this point it is the best I could come up with (events are one tricky subject! and are driving me little insane at this point) and maybe it could help you finding the best possible solution.

This of course does not fix the problems with the overlap from #8786! Just in case you wondered.

**Comment by leofeyer** [comment]
Fixed in 03d2a8b08dad2c30601058d8549a322a103e36a8.

**Comment by arno1979** [comment]
Hi,

I just applied your fix and noticed one little problem:
I included the condition `!$this->cal_noSpan` in my if statements. You chose to leave out this condition and now combined entries for multi-day events that started in the past do not show up in list views with upcoming dates anymore.
Can you please reconsider changing your code to:
```php
if (!$this->cal_noSpan && strncmp($this->cal_format, 'next_', 5) === 0 && $day < $intStart)
{
    continue;
}

if (!$this->cal_noSpan && strncmp($this->cal_format, 'past_', 5) === 0 && $day >= $intEnd)
{
    continue;
}
``` 

Thanks!

**Comment by intradesign** [comment]
Hi,
is it possible that there is a similar problem with recurring events?

It seems that recurring events in the past show up in event lists when the event list settings are set to "show future events". The first occurance of the event seems to be the indicator for showing or not showing the event items.

Thanks!

**Comment by leofeyer** [comment]
> combined entries for multi-day events that started in the past do not show up in shortened list views with upcoming dates anymore

@arno1979 How can I reproduce this?

**Comment by leofeyer** [comment]
Never mind, I found out myself. Fixed in 477bb59146ad61d235886c5c77e36f637e514d3f.
